src/api/posts.ts

```diff
@@ -0,0 +1,7 @@
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+export const getUserPosts = (userId: Post['userId']) => {
```

Comment:
Consider adding error handling to the `getUserPosts` function to manage potential API request failures gracefully. This will help in providing feedback to the user in case of network issues or server errors.